FitGoal/FitGoal/Singup&Singin/Authenticator.swift

```diff
@@ -195,28 +231,46 @@ private enum MissingUserInfoError: Error {
     case noAuthenticationObjectFound
 }
 
-enum SocialMedia: String, CaseIterable {
+enum AuthenticationMethod {
+    case socialMedia(SocialMedia)
+    case custom(CustomAuthentication)
+}
+
+enum SocialMedia {
+    case facebook
+    case google
+    case twitter
+}
+
+struct CustomAuthentication {
+    let name: String
+    let email: String
+    let password: String
+}
+
+enum Provider: String, CaseIterable {
     case facebook = "facebook.com"
     case google = "google.com"
     case twitter = "twitter.com"
+    case emailAndPassword = "password"
 }
 
-private enum AuthenticationMethod {
+private enum ProviderSpecifications {
```

Comment:
Why do you need to change the name? The old one looks better to be than the new one.